hetmech/matrix.py

```diff
@@ -69,5 +76,8 @@ def normalize(matrix, vector, axis, damping_exponent):
         vector **= -damping_exponent
     vector[numpy.isinf(vector)] = 0
     shape = (len(vector), 1) if axis == 'rows' else (1, len(vector))
-    matrix *= vector.reshape(shape)
+    if sparse.issparse(matrix):
+        matrix = matrix.multiply(vector.reshape(shape))
```

Comment:
`.multiply` is only a method for sparse matrices. The two methods in the `if` [...] `else` give the same resulting matrices, but the type of each will be the same as the type of input matrix.